cmake/hip_helpers.cmake

```diff
@@ -1,11 +1,15 @@
 macro(ginkgo_hip_ban_link_hcflag target)
-    get_target_property(GINKGO_TARGET_ILL ${target} INTERFACE_LINK_LIBRARIES)
-    string(REPLACE "-hc " "" GINKGO_TARGET_NEW_ILL "${GINKGO_TARGET_ILL}")
-    set_target_properties(${target} PROPERTIES INTERFACE_LINK_LIBRARIES "${GINKGO_TARGET_NEW_ILL}")
+    if(TARGET ${target})
```

Comment:
Can you explain what this change is necessary for?

Reply:
I can add a comment if needed in the code. In general, it's due to this comment in the PR text:

The MI50 machine available at ICL's cluster uses hip-clang as a compiler.
hip-clang actually fixes some of the most horrible thing with the classical
hcc compiler, in particular it's not necessary to ban flags anymore
because all of these were hcc related.

Reply:
I guess my question was a bit unspecific. I was especially wondering about the condition `TARGET ${target}`. But now that I looked at the call site, it makes perfect sense.

Reply:
To maybe give more detail, if the target doesn't exist as is the case with hip-clang, since there is no more `hcc` and related targets linked, this will do nothing. Otherwise, this banning process would create an error if left without protection.

Right now, this code is purely equivalent to a
```cmake
if (GINKGO_HIP_VERSION GREATER "3.5)
   # old code
endif()
```